tardis/adapters/sites/moab.py

```diff
@@ -82,13 +82,25 @@ def site_name(self):
     async def terminate_resource(self, resource_attributes):
         async with asyncssh.connect(self._remote_host, username=self._login, client_keys=[self._key]) as conn:
             request_command = f"canceljob {resource_attributes.resource_id}"
-            response = await conn.run(request_command, check=True)
-        pattern = re.compile(r'^job \'(\d*)\' cancelled', flags=re.MULTILINE)
+            response = await conn.run(request_command)
         logging.debug(f"{self.site_name} servers terminate returned {response}")
-        resource_id = int(pattern.findall(response.stdout)[0])
-        if resource_id != resource_attributes.resource_id:
-            raise TardisError(f'Failed to terminate {resource_attributes.resource_id}.')
-        resource_attributes.update(resource_status=ResourceStatus.Stopped, updated=datetime.now())
+        if response.exit_status == 0:
+            pattern = re.compile(r'^job \'(\d*)\' cancelled', flags=re.MULTILINE)
+            resource_id = int(pattern.findall(response.stdout)[0])
+            if resource_id != resource_attributes.resource_id:
+                raise TardisError(f'Failed to terminate {resource_attributes.resource_id}.')
```

Comment:
This will crash TARDIS, Is this intended behaviour?

Reply:
The `resource_id` has to be identical. If they are not identical, TARDIS should stop.

Reply:
Agreed